src/ansys/dpf/core/data_tree.py

```diff
@@ -535,18 +537,18 @@ def get_sub_tree_names(self):
         return coll_obj.get_integral_entries()
 
     def __to_dict(self, dic):
-        for attribute_name in self.get_attribute_names():
+        for attribute_name in self.get_attribute_names:
             dic[attribute_name] = self.get_as(attribute_name)
 
-        for sub_tree_name in self.get_sub_tree_names():
+        for sub_tree_name in self.get_sub_tree_names:
             sub_tree = self.get_as(sub_tree_name, types.data_tree)
             sub_dic = {}
             sub_tree.__to_dict(sub_dic)
             dic[sub_tree_name] = sub_dic
 
     def to_dict(self):
```

Comment:
```suggestion
    @property
    def dict(self):
```
You can make this a property
You could also define the `__dict__` dunder method so that `dict(your_object)` works

Reply:
Also, I'd personally integrate `__to_dict` in the `dict` property, and then call directly the `dict` property in other parts of the code.

Reply:
@PProfizi Will do! Wasn't aware of this!